# -*- Mode: perl; indent-tabs-mode: nil -*-
#
# The contents of this file are subject to the Mozilla Public
# License Version 1.1 (the "License"); you may not use this file
# except in compliance with the License. You may obtain a copy of
# the License at http://www.mozilla.org/MPL/
#
# Software distributed under the License is distributed on an "AS
# IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
# implied. See the License for the specific language governing
# rights and limitations under the License.
#
# The Original Code is the TreeView Bugzilla Extension.
#
# The Initial Developer of the Original Code is "Nokia Corpodation"
# Portions created by the Initial Developer are Copyright (C) 2011 the
# Initial Developer. All Rights Reserved.
#
# Contributor(s):
#   Eero Heino <eero.heino@nokia.com>


diff --git a/enter_bug.cgi b/enter_bug.cgi
index c03a14e..16945dc 100755
--- a/enter_bug.cgi
+++ b/enter_bug.cgi
@@ -402,6 +404,7 @@ $vars->{'cloned_bug_id'}         = $cloned_bug_id;
 
 $vars->{'token'}             = issue_session_token('createbug:');
 
+Bugzilla::Hook::process("enter_bug_url_fields", { vars => $vars });
 
 my @enter_bug_fields = grep { $_->enter_bug } Bugzilla->active_custom_fields;
 foreach my $field (@enter_bug_fields) {
@@ -483,6 +486,8 @@ if ($cloned_bug_id) {
         $vars->{$field_name} = $cloned_bug->$field_name;
     }
 
+    Bugzilla::Hook::process("enter_bug_url_fields_cloned", { vars => $vars });
+
     # We need to ensure that we respect the 'insider' status of
     # the first comment, if it has one. Either way, make a note
     # that this bug was cloned from another bug.
